### PR TITLE
MessageConsumer#endHandler should be invoked with the registration context

### PR DESCRIPTION
**Pull request by vietj**
@purplefox please review this fix

### CODE REVIEW

File: src/main/java/io/vertx/core/eventbus/impl/EventBusImpl.java [hunk 1]
```diff
@@ -1150,7 +1150,17 @@ public synchronized boolean isRegistered() {
 
     @Override
     public synchronized MessageConsumer<T> endHandler(Handler<Void> endHandler) {
-      this.endHandler = endHandler;
+      if (endHandler != null) {
+        // We should use the HandlerHolder context to properly do this (needs small refactoring)
```
Comment (purplefox) [inline src/main/java/io/vertx/core/eventbus/impl/EventBusImpl.java]: I agree. So why not do that? ;)

Comment (vietj) [inline src/main/java/io/vertx/core/eventbus/impl/EventBusImpl.java]: I prefer to postpone until next week and open an issue to do be done carefully.

Comment (purplefox) [inline src/main/java/io/vertx/core/eventbus/impl/EventBusImpl.java]: ok fair enough. don't forget though ;)

Comment (vietj) [inline src/main/java/io/vertx/core/eventbus/impl/EventBusImpl.java]: that being said you can do it if you can

Comment (vietj) [inline src/main/java/io/vertx/core/eventbus/impl/EventBusImpl.java]: I won't forget

**Pull request merged**